src/fu-util.c

```diff
@@ -2243,8 +2243,19 @@ fu_util_hwids (FuUtilPrivate *priv, gchar **values, GError **error)
 		NULL };
 
 	/* read DMI data */
-	if (!fu_smbios_setup (smbios, error))
+	if (g_strv_length (values) == 0) {
+		if (!fu_smbios_setup (smbios, error))
+			return FALSE;
+	} else if (g_strv_length (values) == 1) {
+		if (!fu_smbios_setup_from_file (smbios, values[0], error))
```

Comment:
I think you'll either want to duplicate this into `fu-tool.c` or move it there (since it's the uncommon use to pass in a file)

Reply:
hwids is only available in `fwupdmgr` -- although I think it better belongs in the fu-tool rather than fu-util, a lot of people are expecting it to exist with the mgr version. Maybe this this is the time to move it?

Reply:
Yes agree

Reply:
Fixed; I've left a stub in place for the OEMs.